### ISSUE TITLE
Electra scheduler and optimizer

### ISSUE THREAD

**Issue by lucadiliello** [iKernels/transformers-lightning#62]
Implementation of ELECTRA `scheduler` with `layerwise learning rate decay` and ELECTRA `optimizer` (a slightly modified version of AdamW). Original code provided in tensorflow here: https://github.com/google-research/electra/blob/master/model/optimization.py
